src/main/java/Parser.java

```diff
@@ -0,0 +1,219 @@
+package seedu.duke;
+
+import seedu.duke.exception.DukeException;
+import seedu.duke.exception.DukeMissingDescriptionException;
+import seedu.duke.exception.DukeDeadlineMissingDateException;
+import seedu.duke.exception.DukeNoSuchCommandException;
+import seedu.duke.exception.DukeEventMissingDateException;
+import seedu.duke.exception.DukeMissingIndexException;
+import seedu.duke.exception.DukeMissingSearchTermException;
+import seedu.duke.exception.DukeInvalidSortParameterException;
+/**
+ * Reads in the input line, and parses accordingly.
+ * Will return null if trying to access information that doesn't exist for that command.
+ * @author Lim Daekoon
+ */
+
+public class Parser {
+
+    private static final String[] ALLOWED_SORT_BY = {"name", "date"};
+    private static final String[] ALLOWED_SORT_ORDER = {"ascending", "descending"};
+
+    private String type;
+    private String description;
+    private String time;
+    private Integer index;
+    private String searchTerm;
+    private String sortBy;
+    private String sortOrder;
+
+    /**
+     * Constructs a Parser object based on input string.
+     * @param input The command String to be parsed.
```

Comment:
JavaDocs: There should be a new line after the method description and before the parameters.